### PR TITLE
expression: implement vectorized evaluation for `builtinGreatestDecimalSig`

### PR DESCRIPTION
**Pull request by Reminiscent**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinGreatestDecimalSig.

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinFunc/builtinGreatestDecimalSig-VecBuiltinFunc-4          50000             40570 ns/op
BenchmarkVectorizedBuiltinFunc/builtinGreatestDecimalSig-NonVecBuiltinFunc-4       30000             55793 ns/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_compare_vec.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalDecimal(b.ctx, input, result); err != nil {
+		return err
+	}
+
+	d64s := result.Decimals()
+	colLength := len(d64s)
```
Comment (qw4990) [inline expression/builtin_compare_vec.go]: It seems that we don't need this variable since `colLength` can be replaced with `n`.

Comment (Reminiscent) [reply expression/builtin_compare_vec.go]: update. PTAL

**Comment by qw4990** [comment]
@Reminiscent Please resolve conflicts.

**Comment by zz-jason** [comment]
to https://github.com/pingcap/tidb/issues/12058

**Pull request merged**